src/Microsoft.AspNetCore.SignalR.Client.TS/Formatters.ts

```diff
@@ -0,0 +1,146 @@
+﻿import { Message, MessageType } from './Message';
+
+let knownTypes = {
+    "T": MessageType.Text,
+    "B": MessageType.Binary,
+    "C": MessageType.Close,
+    "E": MessageType.Error
+};
+
+function splitAt(input: string, searchString: string, position: number): [string, number] {
+    let index = input.indexOf(searchString, position);
+    if (index < 0) {
+        return [input.substr(position), input.length];
+    }
+    let left = input.substring(position, index);
+    return [left, index + searchString.length];
+}
+
+export namespace SSE {
```

Comment:
I was wondering why not to make these parsefunctions members of the actual transports. At least for ServerSentEvents it  is already coupled to the transport so I am not sure what we gain by keeping it separate. 

I also thought about making them actual classes. In case of long polling you could keep state inside the parser itself instead of outside. You also wouldn't need to return tuples. On the flip side the majority of poll payloads would still contain just one message and we would create a new parser to parse each of these.